### PR TITLE
Adding filesystem metrics

### PR DESCRIPTION
**Pull request by vishh**
Note that support in cadvisor for docker container fs metrics is not complete. 

@vmarmol @rjnagal: I have chosen 'resource_id' as the label to uniquely identify sub-resource types like network cards and filesystem partitions. Any thoughts?

### CODE REVIEW

**Comment by vmarmol** [comment]
CI error:

```
$ go vet github.com/GoogleCloudPlatform/heapster/...
sources/events.go:110: result of fmt.Errorf call not used
exit status 1
```

File: docs/storage-schema.md [hunk 1]
```diff
@@ -26,6 +27,7 @@ Heapster tags each metric with the following labels.
 | container_name | User-provided name of the container or full cgroup name for system containers | v0.9            | No                  |
 | labels         | Comma-separated list of user-provided labels. Format is 'key:value'           | v0.9            | Yes                 |
 | hostname       | Hostname where the container ran                                              | v0.9            | No                  |
+| resource_id    | An unique identifier used to differentiate multiple metrics of the same type. e.x. fs partitions under filesystem/usage | HEAD | No |
```
Comment (vmarmol) [inline docs/storage-schema.md]: nit: s/fs/FS/

File: sinks/api/supported_metrics.go [hunk 2]
```diff
@@ -152,12 +152,36 @@ var statMetrics = []SupportedStatMetric{
...
+			Description: "Total number of bytes consumed on a filesystem",
+			Type:        MetricGauge,
+			ValueType:   ValueInt64,
+			Units:       UnitsCount,
+		},
+		HasValue: func(spec *cadvisor.ContainerSpec) bool {
+			return spec.HasFilesystem
+		},
+		GetValue: func(spec *cadvisor.ContainerSpec, stat *cadvisor.ContainerStats) []internalPoint {
+			result := []internalPoint{}
```
Comment (vmarmol) [inline sinks/api/supported_metrics.go]: nit: make([]internalPoint, 0, len(stats.Filesystem))

Sorry, I know I'm a pain :)

File: sinks/api/types.go [hunk 3]
```diff
@@ -129,6 +129,11 @@ type Point struct {
 	Value interface{}
 }
 
+type internalPoint struct {
+	labels map[string]string
```
Comment (vmarmol) [inline sinks/api/types.go]: Can we document auto-population of these? I'm guessing we only set the specified labels and may set the values of the rest

**Comment by vmarmol** [comment]
LGTM, minor nits. Thanks for getting this in so quickly :)

File: sinks/api/supported_metrics.go (hunk 2 above)
Comment (vishh) [inline sinks/api/supported_metrics.go]: Good suggestion

File: sinks/api/types.go (hunk 3 above)
Comment (vishh) [inline sinks/api/types.go]: Added some documentation.

File: docs/storage-schema.md (hunk 1 above)
Comment (vishh) [inline docs/storage-schema.md]: Done

**Comment by vishh** [comment]
@vmarmol: Fixed your comments. PTAL

**Comment by vmarmol** [comment]
@vishh 

```
--- FAIL: TestHeapsterInfluxDBWorks (9.84 seconds)
	assertions.go:148: 
                        
	Location:	kube_test.go:370
		
	Error:		No error is expected but got failed to build go binary ("exit status 1") - "# github.com/GoogleCloudPlatform/heapster/sinks/api\n../../sinks/api/supported_metrics.go:171: undefined: stats\ngodep: go exit status 2\n"
		
	Messages:	failed to build and push images
```

File: sinks/api/types.go [hunk 4]
```diff
@@ -129,6 +129,14 @@ type Point struct {
 	Value interface{}
 }
 
+// internalPoint is an internal object.
+type internalPoint struct {
+	// Overrides any default labels generated for every Point.
+	// This is typically used for metrics specific labels like 'resource_id'.
```
Comment (vmarmol) [inline sinks/api/types.go]: nit: "metric specific" (without the s)

**Comment by vmarmol** [comment]
LGTM overall

File: sinks/api/types.go (hunk 4 above)
Comment (vishh) [inline sinks/api/types.go]: Done

**Comment by vmarmol** [comment]
LGTM :) merging!

**Pull request merged**